libs/common/src/state-migrations/migrations/39-move-key-connector-to-state-provider.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { MockProxy } from "jest-mock-extended";
+
+import { KeyDefinitionLike, MigrationHelper } from "../migration-helper";
+import { mockMigrationHelper } from "../migration-helper.spec";
+
+import { KeyConnectorMigrator } from "./39-move-key-connector-to-state-provider";
+
+function exampleJSON() {
+  return {
+    profile: {
+      usesKeyConnector: true,
+      convertAccountToKeyConnector: true,
+      otherProfileStuff: "profile.otherStuff1",
+    },
+  };
+}
+
+function rollbackJSON() {
+  return {
+    global_keyConnector_usesKeyConnector: false,
+    global_keyConnector_convertAccountToKeyConnector: false,
+    global: {
+      otherGlobalStuff: "global.otherStuff1",
+    },
+    profile: {
+      otherProfileStuff: "profile.otherStuff1",
+    },
+    otherStuff: "otherStuff2",
+  };
+}
+
+const usesKeyConnectorKeyDefinition: KeyDefinitionLike = {
+  key: "usesKeyConnector",
+  stateDefinition: {
+    name: "keyConnector",
+  },
+};
+
+const convertAccountToKeyConnectorKeyDefinition: KeyDefinitionLike = {
+  key: "convertAccountToKeyConnector",
+  stateDefinition: {
+    name: "keyConnector",
+  },
+};
+
+describe("KeyConnectorMigrator", () => {
+  let helper: MockProxy<MigrationHelper>;
+  let sut: KeyConnectorMigrator;
+
+  describe("migrate", () => {
+    beforeEach(() => {
+      helper = mockMigrationHelper(exampleJSON(), 38);
+      sut = new KeyConnectorMigrator(38, 39);
+    });
+
+    it("should remove usesKeyConnector and convertAccountToKeyConnector from Profile", async () => {
+      await sut.migrate(helper);
+
+      expect(helper.set).toHaveBeenCalledTimes(2);
+      expect(helper.set).toHaveBeenCalledWith("profile", {
+        otherProfileStuff: "profile.otherStuff1",
+      });
+    });
+
+    it("should set usesKeyConnector globally", async () => {
+      await sut.migrate(helper);
+
+      expect(helper.setToGlobal).toHaveBeenCalledTimes(2);
+      expect(helper.setToGlobal).toHaveBeenCalledWith(usesKeyConnectorKeyDefinition, true);
+    });
+
+    it("should set convertAccountToKeyConnector globally", async () => {
+      await sut.migrate(helper);
+
+      expect(helper.setToGlobal).toHaveBeenCalledTimes(2);
+      expect(helper.setToGlobal).toHaveBeenCalledWith(
+        convertAccountToKeyConnectorKeyDefinition,
+        true,
+      );
+    });
+  });
+
+  describe("rollback", () => {
+    beforeEach(() => {
+      helper = mockMigrationHelper(rollbackJSON(), 38);
```

Comment:
Last I talked with @justindbaur we were doing latest on rollback here:

```suggestion
      helper = mockMigrationHelper(rollbackJSON(), 39);
```

Not blocking, just curious about what convention we use here, if any

Reply:
Why would we want to rollback to the current? 🤔